pkg/streamingpromql/operators/aggregations/common.go

```diff
@@ -16,22 +16,23 @@ import (
 // AggregationGroup accumulates series that have been grouped together and computes the output series data.
 type AggregationGroup interface {
 	// AccumulateSeries takes in a series as part of the group
-	AccumulateSeries(data types.InstantVectorSeriesData, timeRange types.QueryTimeRange, memoryConsumptionTracker *limiting.MemoryConsumptionTracker, emitAnnotationFunc types.EmitAnnotationFunc) error
+	AccumulateSeries(data types.InstantVectorSeriesData, timeRange types.QueryTimeRange, memoryConsumptionTracker *limiting.MemoryConsumptionTracker, emitAnnotationFunc types.EmitAnnotationFunc, remainingSeriesToGroup uint) error
```

Comment:
[nit] Might be good to document that `remainingSeriesToGroup` includes the series being passed to `AccumulateSeries` (eg. if this is the last series, then `remainingSeriesToGroup` is 1)